pkg/controllermanager/controller/seed/seed_reconcile.go

```diff
@@ -0,0 +1,204 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package seed
+
+import (
+	"context"
+	"fmt"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	gardencorelisters "github.com/gardener/gardener/pkg/client/core/listers/core/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/client/kubernetes/clientmap"
+	"github.com/gardener/gardener/pkg/client/kubernetes/clientmap/keys"
+	"github.com/gardener/gardener/pkg/logger"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (c *Controller) seedEnqueue(obj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		return
+	}
+
+	c.seedQueue.Add(key)
+}
+
+func (c *Controller) seedAdd(obj interface{}) {
+	seed, ok := obj.(*gardencorev1beta1.Seed)
+	if !ok {
+		return
+	}
+
+	c.seedEnqueue(seed)
+}
+
+type reconciler struct {
+	clientMap    clientmap.ClientMap
+	secretLister corev1listers.SecretLister
+	seedLister   gardencorelisters.SeedLister
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	seedObj, err := r.seedLister.Get(req.Name)
+	if apierrors.IsNotFound(err) {
+		logger.Logger.Infof("[SEED] Stopping operations for Seed %s since it has been deleted", req.Name)
+		return reconcileResult(nil)
+	}
+	if err != nil {
+		logger.Logger.Infof("[SEED] %s - unable to retrieve object from store: %v", req.Name, err)
+		return reconcileResult(err)
+	}
+
+	gardenClient, err := r.clientMap.GetClient(ctx, keys.ForGarden())
+	if err != nil {
+		return reconcileResult(fmt.Errorf("failed to get garden client: %w", err))
+	}
+
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("seed-%s", seedObj.Name),
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(seedObj, gardencorev1beta1.SchemeGroupVersion.WithKind("Seed")),
+			},
+		},
+	}
+
+	// TODO: use controllerutil.CreateOrUpdate once cache is graduated to beta https://github.com/gardener/gardener/issues/2822
+	if err := gardenClient.Client().Create(ctx, namespace); err != nil {
```

Comment:
Wouldn't you use the DirectClient() here then?

Reply:
There is no cache interaction for client write operations. @timebertt are you aware of the opposite?

Reply:
> @timebertt are you aware of the opposite?

Nope. I think so, too.